aten/src/ATen/core/jit_type.h

```diff
@@ -1777,6 +1777,24 @@ struct CAFFE2_API ClassType : public NamedType {
     return parameterSlots_->at(slot);
   }
 
+  void addForwardPreHook(QualifiedName hook_name) {
```

Comment:
methods on the class is stored as `std::vector<torch::jit::Function*> methods_;`

is there a reason not to store the hook as a method too ?

Reply:
This is more about storing which methods are hooks rather than storing the methods themselves. This information is later used to either dump the info into the saved model or use the saved model to create an `nn.Module` representation and set the hooks there correctly.